model/clusters_mgmt/v1/component_route_kind_type.model

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Type of Component Route.
+enum ComponentRouteType {
+	// Oauth.
+	@json(name = "oauth")
```

Comment:
Do we need the json field mapping? I think it would be the default as the field name in lowercase.